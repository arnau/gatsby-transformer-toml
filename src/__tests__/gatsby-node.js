const { onCreateNode } = require('../gatsby-node')

// Copied from
// https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-transformer-toml
describe('Process TOML nodes correctly', () => {
  const node = {
    id: 'whatever',
    parent: null,
    children: [],
    extension: 'toml',
    internal: {
      contentDigest: 'whatever',
    },
    name: 'test',
  }

  // Provide fake functions
  const loadNodeContent = node => Promise.resolve(node.content)

  it('Correctly creates nodes from TOML test file', async () => {
    node.content = `
    title = "a title"
    date = 2019-07-20
    notes = ['A', 'C', 'C', 'E', 'B']

    [[editors]]
    name = "Alice"
    url = "~alicewonder"

    [[editors]]
    name = "Bob"
    url = "~bobsponge"
    `

    const createNode = jest.fn()
    const createParentChildLink = jest.fn()
    const actions = { createNode, createParentChildLink }
    const createNodeId = jest.fn()
    createNodeId.mockReturnValue('uuid-from-gatsby')
    const createContentDigest = jest.fn().mockReturnValue('contentDigest')

    await onCreateNode({
      node,
      loadNodeContent,
      actions,
      createNodeId,
      createContentDigest,
    }).then(() => {
      expect(createNode.mock.calls).toMatchSnapshot()
      expect(createParentChildLink.mock.calls).toMatchSnapshot()
      expect(createNode).toHaveBeenCalledTimes(1)
      expect(createParentChildLink).toHaveBeenCalledTimes(1)
      expect(createContentDigest).toHaveBeenCalledTimes(1)
    })
  })
})
