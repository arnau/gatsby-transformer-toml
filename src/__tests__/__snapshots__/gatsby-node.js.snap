// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Process TOML nodes correctly Correctly creates nodes from TOML test file 1`] = `
Array [
  Array [
    Object {
      "children": Array [],
      "date": 2019-07-20T00:00:00.000Z,
      "editors": Array [
        Object {
          "name": "Alice",
          "url": "~alicewonder",
        },
        Object {
          "name": "Bob",
          "url": "~bobsponge",
        },
      ],
      "id": "uuid-from-gatsby",
      "internal": Object {
        "content": "
    title = \\"a title\\"
    date = 2019-07-20
    notes = ['A', 'C', 'C', 'E', 'B']

    [[editors]]
    name = \\"Alice\\"
    url = \\"~alicewonder\\"

    [[editors]]
    name = \\"Bob\\"
    url = \\"~bobsponge\\"
    ",
        "contentDigest": "contentDigest",
        "description": "A Toml based node of type TOML",
        "mediaType": "application/toml",
        "type": "TOML",
      },
      "notes": Array [
        "A",
        "C",
        "C",
        "E",
        "B",
      ],
      "parent": "whatever",
      "title": "a title",
    },
  ],
]
`;

exports[`Process TOML nodes correctly Correctly creates nodes from TOML test file 2`] = `
Array [
  Array [
    Object {
      "child": Object {
        "children": Array [],
        "date": 2019-07-20T00:00:00.000Z,
        "editors": Array [
          Object {
            "name": "Alice",
            "url": "~alicewonder",
          },
          Object {
            "name": "Bob",
            "url": "~bobsponge",
          },
        ],
        "id": "uuid-from-gatsby",
        "internal": Object {
          "content": "
    title = \\"a title\\"
    date = 2019-07-20
    notes = ['A', 'C', 'C', 'E', 'B']

    [[editors]]
    name = \\"Alice\\"
    url = \\"~alicewonder\\"

    [[editors]]
    name = \\"Bob\\"
    url = \\"~bobsponge\\"
    ",
          "contentDigest": "contentDigest",
          "description": "A Toml based node of type TOML",
          "mediaType": "application/toml",
          "type": "TOML",
        },
        "notes": Array [
          "A",
          "C",
          "C",
          "E",
          "B",
        ],
        "parent": "whatever",
        "title": "a title",
      },
      "parent": Object {
        "children": Array [],
        "content": "
    title = \\"a title\\"
    date = 2019-07-20
    notes = ['A', 'C', 'C', 'E', 'B']

    [[editors]]
    name = \\"Alice\\"
    url = \\"~alicewonder\\"

    [[editors]]
    name = \\"Bob\\"
    url = \\"~bobsponge\\"
    ",
        "extension": "toml",
        "id": "whatever",
        "internal": Object {
          "contentDigest": "whatever",
        },
        "name": "test",
        "parent": null,
      },
    },
  ],
]
`;
